website/project/model.py

```diff
@@ -248,13 +288,16 @@ def create(cls, auth, **kwargs):
         else:
             comment.root_target = comment.target
 
-        if isinstance(comment.root_target.referent, Node):
-            comment.page = Comment.OVERVIEW
-        elif isinstance(comment.root_target.referent, StoredFileNode):
-            log_dict['file'] = {'name': comment.root_target.referent.name, 'url': comment.get_comment_page_url()}
-            comment.page = Comment.FILES
-        else:
+        page = getattr(comment.root_target.referent, 'root_target_page', None)
+        if not page:
             raise ValueError('Invalid root target.')
+        comment.page = page
+
+        if comment.page == Comment.FILES:
```

Comment:
Rather than these conditionals, could you expose a hook on `Commentable`?

```python
class StoredFileNode(..., Commentable):
    def create_log_metadata(self, comment):
        """Return extra data to pass as ``params`` to ``Node.add_log`` when a new comment is created."""
        return {'file': {'name': self.name, 'url': comment.get_comment_page_url()}}
```